cmake/Modules/FindMKLDNN.cmake

```diff
@@ -18,6 +18,47 @@ IF(NOT MKLDNN_FOUND)
 
   SET(IDEEP_ROOT "${PROJECT_SOURCE_DIR}/third_party/ideep")
   SET(MKLDNN_ROOT "${PROJECT_SOURCE_DIR}/third_party/ideep/mkl-dnn")
+
+  if(USE_XPU) # Build oneDNN GPU library
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```

Comment:
What's the problem if we do not add this check?

Reply:
oneDNN build logic for XPU requires passing the HOST_COMPILER to its cmake, and it is xpu-specific logic.

- `USE_XPU` intends to bypass the logic for CPU
- `CMAKE_CXX_COMPILER_ID` intends to decide which HOST_COMPILER we should pass to oneDNN CMake.

Reply:
But why can't we use the default logic but have to check "GNU" specifically?

Reply:
Hi, @jgong5 oneDNN could only parse string `g++` or `/path/to/g++` and `DEFAULT` three options.  When using default options, no explicit host-compiler is passed to icpx, and icpx would use whatever the host compiler it has as a default one.
 
The ideal way for me to pass host-compiler to oneDNN is using `CMAKE_CXX_COMPILER` . But, on a linux system, `CMAKE_CXX_COMPILER` may not the one oneDNN wants, like `/usr/bin/cxx` as I mentioned in comments.

Reply:
I may need further check on the behavior of  `DEFAULT` options. 
I think what we need is that oneDNN chose a host compiler that is same as `${CMAKE_CXX_COMPILER}`. I am not sure  whether oneDNN has the possibility to use `clang` even if the PyTorch uses a gnu compiler.